File: scripts/generate-intro.mjs
// One-time voice generation. Run locally, commit the resulting MP3, done.
// The live site never calls any API — it just plays the stored file.
//
//   GROQ_API_KEY=your_key node scripts/generate-intro.mjs
//
// Get a free key at https://console.groq.com. You may need to accept the
// model terms for `playai-tts` once in the Groq console before this works.
//
// Prefer a different voice engine? Microsoft Edge TTS and Piper are also free
// and produce an MP3 the exact same way — only this script changes.

import { writeFile, mkdir } from "node:fs/promises";

const API_KEY = process.env.GROQ_API_KEY;
if (!API_KEY) {
  console.error("Missing GROQ_API_KEY. Run: GROQ_API_KEY=xxx node scripts/generate-intro.mjs");
  process.exit(1);
}

// Edit this line to taste — keep it to ~2 sentences.
const SCRIPT =
  "Hi, I'm Anushka Maheshwari. I engineer intelligent systems that turn data into decisions.";

const res = await fetch("https://api.groq.com/openai/v1/audio/speech", {
  method: "POST",
  headers: {
    Authorization: `Bearer ${API_KEY}`,
    "Content-Type": "application/json",
  },
  body: JSON.stringify({
    model: "playai-tts",
    voice: "Celeste-PlayAI",
    input: SCRIPT,
    response_format: "mp3",
  }),
});

if (!res.ok) {
  console.error(`Groq TTS failed (${res.status}):`, await res.text());
  process.exit(1);
}

await mkdir("public/audio", { recursive: true });
const buffer = Buffer.from(await res.arrayBuffer());
await writeFile("public/audio/intro.mp3", buffer);
console.log("Saved public/audio/intro.mp3 (" + buffer.length + " bytes)");

File: scripts/generate_intro.py
"""
Generate the intro voice ONCE, then commit public/audio/intro.mp3.
The live website only plays that file — it never calls any API.

Setup:
    pip install -r requirements.txt
    # put your key in .env  ->  GROQ_API_KEY=gsk_...
    python scripts/generate_intro.py

Get a free key at https://console.groq.com (you may need to accept the
playai-tts model terms once in the console).
"""

import os
from pathlib import Path

from dotenv import load_dotenv
from groq import Groq

# Load secrets from the .env file (not from the shell environment).
load_dotenv()

api_key = os.getenv("GROQ_API_KEY")
if not api_key:
    raise SystemExit("GROQ_API_KEY not found. Add it to your .env file.")

# Edit this line to taste — keep it to about two sentences.
SCRIPT = "Hi, I'm Anushka Maheshwari. I engineer intelligent systems that turn data into decisions."

OUTPUT = Path("public/audio/intro.mp3")
OUTPUT.parent.mkdir(parents=True, exist_ok=True)

client = Groq(api_key=api_key)

response = client.audio.speech.create(
    model="playai-tts",
    voice="Celeste-PlayAI",   # browse other voices in the Groq docs
    input=SCRIPT,
    response_format="mp3",
)

response.write_to_file(OUTPUT)
print(f"Saved {OUTPUT} ({OUTPUT.stat().st_size} bytes)")
